### DISCUSSION TITLE
why my website is not indexed on google? (NextJs)

### DISCUSSION THREAD

**Discussion by MaxVhanamane** [vercel/next.js#44988] [Help]
I have created a website using nextjs and when I use a link directly(https://xyz.app) I can access it without any problem. but when I use site:https://xyz.vercel.app to check wheather it is indexed on google or not ,It doesn't show there. What is the problem here. and how many free apps on vercel are indexed. I have another app that is indexed properly i.e site:https://zzz.vercel.app , This is the first app that I deployed on vercel. later I have deployed multiple apps but they don't show anything when I use site: .

**Comment by ciruz** [comment]
Hey @MaxVhanamane,

there is nothing really wrong with your website, but there are some basic SEO topics not right.

I'll give you some really basic SEO tips to get you started:

1. checkout and use https://search.google.com/ - you can ask google to recrawl your url there, or get a report about any issues found by google on your website (it was once called webmaster tools)
2. learn about ``robots.txt``
3. learn about ``sitemap.xml``, it makes sense for your website 
4. learn about the different HTML headings ``<h1>`` - ``<h6>`` and how to use them correctly. For ex. only use one ``<h1>`` on a site, you have three.
5. checkout the different meta tags, your meta description is basically not really a description of your site. It's a list of all dishes.

**Reply by MaxVhanamane** [reply]
Ok, I'll make these changes and see how it goes.

**Reply by MaxVhanamane** [reply]
I made the above changes, but still not showing anything when I use the site: https://redtomato.vercel.app.  I have another app (site:https://shejalesir.vercel.app) and there I haven't added robot.txt and sitemap.xml but still it is indexed on google. That was the first app that I hosted on vercel. Do you know how many free apps  vercel indexes on google.

**Reply by ciruz** [reply]
> I made the above changes, but still not showing anything when I use the site: https://redtomato.vercel.app.  I have another app (site:https://shejalesir.vercel.app) and there I haven't added robot.txt and sitemap.xml but still it is indexed on google. That was the first app that I hosted on vercel. Do you know how many free apps  vercel indexes on google.

Changing the structure and SEO of a site has no immediately effect. It takes days or weeks until Google will visit your site again and also until there is nothing changed in the Google search results. When Google visits your site for the first time and thinks: "meh boring" it will not visit you again on the next day. This takes time, maybe weeks.

Use the Google Search Console to monitor your site for tips and problems. You can ask the Google Bot there to recrawl again or check a specific URL for problems.

Also there are no limitations on Vercel how many sites Google can visit, crawl or index.

**Reply by MaxVhanamane** [reply]
Ok, thanks.
